crates/tabby/src/services/completion.rs

```diff
@@ -109,19 +109,28 @@ pub struct Segments {
     suffix: Option<String>,
 
     /// The relative path of the file that is being edited.
-    /// - When `git_url` is set, this is the path of the file in the git repository.
-    /// - When `git_url` is empty, this is the path of the file in the workspace.
+    /// - When [Segments::git_url] is set, this is the path of the file in the git repository.
+    /// - When [Segments::git_url] is empty, this is the path of the file in the workspace.
     filepath: Option<String>,
 
     /// The remote URL of the current git repository.
     /// Leave this empty if the file is not in a git repository,
     /// or the git repository does not have a remote URL.
     git_url: Option<String>,
 
-    /// The relevant declaration code snippets provided by editor.
-    /// It'll contains declarations extracted from `prefix` segments using LSP.
+    /// The relevant declaration code snippets provided by the editor's LSP,
+    /// contain declarations of symbols extracted from [Segments::prefix].
     declarations: Option<Vec<Declaration>>,
 
+    /// The relevant code snippets extracted from recently accessed files.
+    /// The current editing file is excluded from the search candidates.
+    ///
+    /// When provided alongside [Segments::declarations], the snippets should
+    /// be deduplicated to avoid duplication with entries in `declarations`.
```

Comment:
you should say "it may overlapped with declarations". Instead of asking how it should be used, API should only describe objective fact.

Reply:
My description was wrong, I want to say the snippets have already been deduplicated.